src/Common/oclBasics.cpp

```diff
@@ -0,0 +1,109 @@
+        CASE_CL_CONSTANT(CL_OUT_OF_HOST_MEMORY)
+        CASE_CL_CONSTANT(CL_PROFILING_INFO_NOT_AVAILABLE)
+        CASE_CL_CONSTANT(CL_MEM_COPY_OVERLAP)
+        CASE_CL_CONSTANT(CL_IMAGE_FORMAT_MISMATCH)
+        CASE_CL_CONSTANT(CL_IMAGE_FORMAT_NOT_SUPPORTED)
+        CASE_CL_CONSTANT(CL_BUILD_PROGRAM_FAILURE)
+        CASE_CL_CONSTANT(CL_MAP_FAILURE)
+        CASE_CL_CONSTANT(CL_MISALIGNED_SUB_BUFFER_OFFSET)
+        CASE_CL_CONSTANT(CL_EXEC_STATUS_ERROR_FOR_EVENTS_IN_WAIT_LIST)
+        CASE_CL_CONSTANT(CL_COMPILE_PROGRAM_FAILURE)
+        CASE_CL_CONSTANT(CL_LINKER_NOT_AVAILABLE)
+        CASE_CL_CONSTANT(CL_LINK_PROGRAM_FAILURE)
+        CASE_CL_CONSTANT(CL_DEVICE_PARTITION_FAILED)
+        CASE_CL_CONSTANT(CL_KERNEL_ARG_INFO_NOT_AVAILABLE)
+        CASE_CL_CONSTANT(CL_INVALID_VALUE)
+        CASE_CL_CONSTANT(CL_INVALID_DEVICE_TYPE)
+        CASE_CL_CONSTANT(CL_INVALID_PLATFORM)
+        CASE_CL_CONSTANT(CL_INVALID_DEVICE)
+        CASE_CL_CONSTANT(CL_INVALID_CONTEXT)
+        CASE_CL_CONSTANT(CL_INVALID_QUEUE_PROPERTIES)
+        CASE_CL_CONSTANT(CL_INVALID_COMMAND_QUEUE)
+        CASE_CL_CONSTANT(CL_INVALID_HOST_PTR)
+        CASE_CL_CONSTANT(CL_INVALID_MEM_OBJECT)
+        CASE_CL_CONSTANT(CL_INVALID_IMAGE_FORMAT_DESCRIPTOR)
+        CASE_CL_CONSTANT(CL_INVALID_IMAGE_SIZE)
+        CASE_CL_CONSTANT(CL_INVALID_SAMPLER)
+        CASE_CL_CONSTANT(CL_INVALID_BINARY)
+        CASE_CL_CONSTANT(CL_INVALID_BUILD_OPTIONS)
+        CASE_CL_CONSTANT(CL_INVALID_PROGRAM)
+        CASE_CL_CONSTANT(CL_INVALID_PROGRAM_EXECUTABLE)
+        CASE_CL_CONSTANT(CL_INVALID_KERNEL_NAME)
+        CASE_CL_CONSTANT(CL_INVALID_KERNEL_DEFINITION)
+        CASE_CL_CONSTANT(CL_INVALID_KERNEL)
+        CASE_CL_CONSTANT(CL_INVALID_ARG_INDEX)
+        CASE_CL_CONSTANT(CL_INVALID_ARG_VALUE)
+        CASE_CL_CONSTANT(CL_INVALID_ARG_SIZE)
+        CASE_CL_CONSTANT(CL_INVALID_KERNEL_ARGS)
+        CASE_CL_CONSTANT(CL_INVALID_WORK_DIMENSION)
+        CASE_CL_CONSTANT(CL_INVALID_WORK_GROUP_SIZE)
+        CASE_CL_CONSTANT(CL_INVALID_WORK_ITEM_SIZE)
+        CASE_CL_CONSTANT(CL_INVALID_GLOBAL_OFFSET)
+        CASE_CL_CONSTANT(CL_INVALID_EVENT_WAIT_LIST)
+        CASE_CL_CONSTANT(CL_INVALID_EVENT)
+        CASE_CL_CONSTANT(CL_INVALID_OPERATION)
+        CASE_CL_CONSTANT(CL_INVALID_GL_OBJECT)
+        CASE_CL_CONSTANT(CL_INVALID_BUFFER_SIZE)
+        CASE_CL_CONSTANT(CL_INVALID_MIP_LEVEL)
+        CASE_CL_CONSTANT(CL_INVALID_GLOBAL_WORK_SIZE)
+        CASE_CL_CONSTANT(CL_INVALID_PROPERTY)
+        CASE_CL_CONSTANT(CL_INVALID_IMAGE_DESCRIPTOR)
+        CASE_CL_CONSTANT(CL_INVALID_COMPILER_OPTIONS)
+        CASE_CL_CONSTANT(CL_INVALID_LINKER_OPTIONS)
+        CASE_CL_CONSTANT(CL_INVALID_DEVICE_PARTITION_COUNT)
+        default:
+            return "UNKNOWN ERROR CODE ";
+    }
+
+#undef CASE_CL_CONSTANT
+}
+
+
+static void checkError(cl_int error)
+{
+    if (error != CL_SUCCESS)
+        throw std::runtime_error("OpenCL error " + opencl_error_to_str(error) + ".");
```

Comment:
We should throw our Exception instead. Add new code entry to Common/ErrorCodes.cpp for it. I.e. `OPENCL_ERROR`